3rdparty/Makefile.inc

```diff
@@ -1,2 +1,3 @@
 THIRDPARTY_DIR = $(dir $(lastword $(MAKEFILE_LIST)))
 include $(THIRDPARTY_DIR)/everest/Makefile.inc
+include ../3rdparty/p256-m/Makefile.inc
```

Comment:
`$(word 2, $(MAKEFILE_LIST))` only works if `3rdparty/Makefile.inc` is included from a makefile that is the one read on the make command line. This is fragile and [caused some head scratching](https://github.com/Mbed-TLS/mbedtls/pull/8283#discussion_r1345635747).

What's going on here is that make is a strange language. Variables are call-by-name — GNU make calls this “[recursively expanded](https://www.gnu.org/software/make/manual/html_node/Recursive-Assignment.html)”. A make variable has an expansion, not a value. After the definition `THIRDPARTY_DIR = stuff`, whenever make sees `$(THIRDPARTY_DIR)`, it evaluates `stuff` again. This is a common feature in declarative languages, which make partly is, and also in macro expansion languages, which make partly is. But it gets confusing when mixed with imperative traits, which make also has. Check this out:

```
THIRDPARTY_DIR = $(dir $(lastword $(MAKEFILE_LIST)))
$(info [[MAKEFILE_LIST = $(MAKEFILE_LIST)]])
include $(THIRDPARTY_DIR)/everest/Makefile.inc
$(info [[MAKEFILE_LIST = $(MAKEFILE_LIST)]])
include $(THIRDPARTY_DIR)/p256-m/Makefile.inc
$(info [[MAKEFILE_LIST = $(MAKEFILE_LIST)]])
```

Each time make parses an included makefile, it appends to the value of `MAKEFILE_LIST` (imperative aspect), and this is reflected in the expansion of `THIRDPARTY_DIR` since it's a recursively expanded variable. The last word changes each time.

The correct fix (which I'll make a patch for) is to use a [simply expanded](https://www.gnu.org/software/make/manual/html_node/Simple-Assignment.html) (call-by-value) variable. You do this with a `:=` assignment. A variable assigned with `:=` has a fixed value, calculated when make sees the assignment.
```
THIRDPARTY_DIR := $(dir $(lastword $(MAKEFILE_LIST)))
include $(THIRDPARTY_DIR)/everest/Makefile.inc
include $(THIRDPARTY_DIR)/p256-m/Makefile.inc
```